feat: add markdown-it-github-headings (#22)

Flet opened the pull request:
- Also add marky-deep-links to the default and site layouts.

Note that this will now add the little svg link icons next to each header (similar to GitHub):
![image](https://user-images.githubusercontent.com/360233/30012679-6582e0e2-9107-11e7-8aff-ec1f28451c20.png)

They show all the time instead of just on hover though... maybe that should be updated in style.css?

Flet commented on index.js:
```diff
@@ -89,6 +90,21 @@ function sitedown (options, callback) {
 function mdToHtml (filePath) {
   var body = fs.readFileSync(filePath, encoding)
+
+  var md = markdownIt(mdOpts)
```
The creation of the markdownIt instance had to be moved here in order to ensure link anchors are generated with the same numbering every time.

[github-slugger](https://github.com/Flet/github-slugger) (which markdown-it-github-headings uses) generates links like GitHub does. If there are duplicate links, a number is appended to the end. getting a new instance of markdownIt ensures that numbering is reset on every run. If we didn't do this, the counters would never reset between uses.

ungoldman replied:
ok, that makes sense

Flet commented on layout.html:
```diff
@@ -8,5 +8,6 @@
 </head>
 <body>
   <main class="markdown-body"></main>
+  <script src="https://unpkg.com/marky-deep-links@1.0.1/marky-deep-links-standalone-min.js"></script>
```
[`marky-deep-links`](https://github.com/Flet/marky-deep-links) is a little snippet of code that will ensure navigating to `#niceanchor` will actually navigate to `#user-content-niceanchor` since all anchors are prefixed to ensure [DOM Clobbering](https://github.com/Flet/markdown-it-github-headings#why-should-i-prefix-heading-ids) does not happen.

ungoldman replied:
I'm not sure I like adding this to the default layout. I think the `marky-deep-links` might make more sense as something the user can add on their own. Prefixing everything with `user-content-` might make sense for a large website with a lot of user generated content, like github or npm, but not necessarily for a small website generated from markdown with `sitedown`.

ungoldman replied:
ah, I understand the need for `marky-deep-links` now. this seems a bit like over-optimization though. my site definitely doesn't need every ID to be prefixed `user-content-`.

While I understand the concern about DOM clobbering, this is more of an issue if the user is using IDs for CSS (which is a bad idea), or for JS (which is generally not a great idea either), and the default use case of sitedown has no CSS and no JS. For that reason I'd rather not include this snippet and the prefixing out of the box by default.

I see the `prefixHeadingIds` is set to true by default here: https://github.com/Flet/markdown-it-github-headings#options-and-defaults -- is there a way to make this optional in sitedown as well?

Flet commented on test/index.js:
```diff
@@ -214,16 +214,21 @@ test('site generation - custom element', function (t) {
+var coolSvgLinkIconWithPathCloseTag = '<svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.6 ...
+var coolSvgLinkIcon = '<svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16"><path d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1 ...
+
+var generatedIndexLgtm = '<h1><a id="user-content-testing" class="anchor" href="#testing" aria-hidden="true">' + coolSvgLinkIconWithPathCloseTag + '</a>TESTING!</h1>\n'
```
I'm not sure why markdown-it does not collapse the `<path>` tag for this run but it does for everything else.

bcomnes commented:
TY I've wanted this for a long while.

Flet commented:
OK! So, now there is no prefix by default and `marky-deep-links` is not referenced. Using the `--prefix` option will add a prefix to heading IDs. I also added a note about prefixing heading IDs in the README.

I also added a couple of tests using the prefix option.

Flet commented:
@ungoldman Thanks again for the chat! Update the PR to add `--github-headings` option that adds un-prefixed anchors (with svg icon) to headings.

bcomnes commented:
Something about how I do things broke my site within semver:

![screen shot 2017-09-06 at 12 28 54 pm](https://user-images.githubusercontent.com/166301/30130721-feb783f0-92fe-11e7-9272-18235fcb1f22.png)

bcomnes commented:
I can look into it later

ungoldman commented:
rut roh

ungoldman commented:
@bcomnes actually this is probably unrelated to @flet's PR, I made some breaking changes in `style.css` recently.

https://github.com/ungoldman/style.css/blob/master/changelog.md#100-beta---2016-09-01

In my defense, it was an alpha release up until 5 days ago.

bcomnes commented:
Oh yeah that makes sense
